src/sync.rs

```diff
@@ -916,11 +929,11 @@ impl<T: Copy + Clone> Clone for LockFreeFrozenVec<T> {
             *coppied_data[buffer_index].get_mut() = new_buffer_ptr;
         });
 
-        return Self {
+        Self {
             data: coppied_data,
-            len: AtomicUsize::new(self.len.load(Ordering::Relaxed)),
+            len: AtomicUsize::new(self.len()),
```

Comment:
issue: this should also stay Relaxed I think

Reply:
oh, I see your comment about stores being Release.

Can you note that in a comment here?